aten/src/ATen/core/jit_type.h

```diff
@@ -1105,12 +1105,18 @@ using ::torch::jit::script::Method;
 
 // This represents a class in TorchScript.
 struct CAFFE2_API ClassType : public Type {
-  // Create a user type and register it globally.
+  static std::string getFreshNamespace();
+
+  // Create a user type and register it in the namespace `ns`
   static ClassTypePtr create(
+      const std::string& ns,
```

Comment:
Symbol already is a qualified name. If we used symbols then we would not have to pass around two strings everywhere to represent one name.

Reply:
Makes sense. I'll move the namespace isolation into Symbol.